File: frontend/src/types/company.ts
export interface Company {
  id: string
  name: string
  career_url: string
  crawl_interval_hours: number
  last_crawled_at: string | null
  job_count: number
  crawl_status: 'idle' | 'pending' | 'running' | 'completed' | 'failed' | 'cancelled'
}

export interface CompanyCreate {
  id: string
  name: string
  career_url: string
  crawl_interval_hours: number
}

export interface CompanyUpdate {
  name?: string
  career_url?: string
  crawl_interval_hours?: number
}

File: backend/app/schemas/company.py
from pydantic import BaseModel


class CompanyCreate(BaseModel):
    id: str
    name: str
    career_url: str
    crawl_interval_hours: int = 12


class CompanyUpdate(BaseModel):
    name: str | None = None
    career_url: str | None = None
    crawl_interval_hours: int | None = None


class CompanyOut(BaseModel):
    id: str
    name: str
    career_url: str
    crawl_interval_hours: int
    last_crawled_at: str | None = None
    job_count: int = 0
    crawl_status: str = "idle"  # idle / pending / running / completed / failed
